src/OpenTelemetry.Contrib.Extensions.AWSXRay/Trace/AWSXRaySamplerClient.cs

```diff
@@ -0,0 +1,107 @@
+// <copyright file="AWSXRaySamplerClient.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace OpenTelemetry.Contrib.Extensions.AWSXRay.Trace;
+internal class AWSXRaySamplerClient : IDisposable
+{
+    private readonly string getSamplingRulesEndpoint;
+    private readonly HttpClient httpClient;
+    private readonly string jsonContentType = "application/json";
+
+    public AWSXRaySamplerClient(string host)
+    {
+        this.getSamplingRulesEndpoint = host + "/GetSamplingRules";
+        this.httpClient = new HttpClient();
+    }
+
+    public async Task<List<SamplingRule>> GetSamplingRules()
+    {
+        List<SamplingRule> samplingRules = new List<SamplingRule>();
+
+        var request = new HttpRequestMessage(HttpMethod.Post, this.getSamplingRulesEndpoint)
+        {
+            Content = new StringContent(string.Empty, Encoding.UTF8, this.jsonContentType),
+        };
+
+        var responseJson = await this.DoRequestAsync(this.getSamplingRulesEndpoint, request).ConfigureAwait(false);
+
+        try
+        {
+            GetSamplingRulesResponse? getSamplingRulesResponse = JsonConvert.DeserializeObject<GetSamplingRulesResponse>(responseJson);
```

Comment:
There is ongoing PR with replacing Newtonsoft.json by System.Text.Json.

Reply:
Thanks for mentioning that. I will look into using `System.Text.Json`.